downstairs/src/lib.rs

```diff
@@ -3316,9 +3312,7 @@ impl Work {
         let num_deps_outstanding = work
```

Comment:
non-blocking: what about `num_deps_outstanding` moving into the helper too?

Reply:
I will consider it; not sure if I want `CompletedJobs` to know about the `IOop`, though...